### PR TITLE
Support --x-dev-env for Pages

### PR DESCRIPTION
**Pull request by RamIdeas**
## What this PR solves / how to test

Fixes [DEVX-1296](https://jira.cfdata.org/browse/DEVX-1296)

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: experimental flag

### CODE REVIEW

File: packages/wrangler/src/pages/dev.ts [hunk 1]
```diff
@@ -680,6 +694,8 @@ export const Handler = async (args: PagesDevArguments) => {
 			showInteractiveDevSession: args.showInteractiveDevSession,
 			testMode: false,
 			watch: true,
+			fileBasedRegistry: args.experimentalRegistry,
```
Comment (penalosa) [inline packages/wrangler/src/pages/dev.ts:697]: This shouldn't be added here—bad rebase maybe?

File: packages/wrangler/src/pages/dev.ts [hunk 2]
```diff
@@ -239,9 +239,23 @@ export function Options(yargs: CommonYargsArgv) {
...
+					"Use the experimental DevEnv instantiation (unified across wrangler dev and unstable_dev)",
+				default: false,
+			},
+			"experimental-registry": {
+				alias: ["x-registry"],
+				type: "boolean",
+				describe:
+					"Use the experimental file based dev registry for multi-worker development",
+				default: false,
+			},
```
Comment (penalosa) [inline packages/wrangler/src/pages/dev.ts:258]: As below

File: packages/wrangler/src/pages/dev.ts [hunk 3]
```diff
@@ -239,9 +239,23 @@ export function Options(yargs: CommonYargsArgv) {
 			},
 			"show-interactive-dev-session": {
 				describe:
-					"Show interactive dev session (defaults to true if the terminal supports interactivity)",
+					"Use the experimental DevEnv instantiation (unified across wrangler dev and unstable_dev)",
```
Comment (penalosa) [inline packages/wrangler/src/pages/dev.ts:242]: Typo

**Review by penalosa — DISMISSED** [review]

File: packages/wrangler/e2e/__snapshots__/pages-dev.test.ts.snap [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  - SERVICE_BINDING_3_TOML: SERVICE_NAME_3_ARGS
+- AI:
+  - Name: AI_BINDING_2_TOML
+- Vars:
+  - VAR1: "(hidden)"
+  - VAR2: "VAR_2_TOML"
+  - VAR3: "(hidden)"
+▲ [WARNING] ⎔ Support for service bindings in local mode is experimental and may change.
+▲ [WARNING] ⎔ Support for external Durable Objects in local mode is experimental and may change.
+(node:75279) ExperimentalWarning: The Ed25519 Web Crypto API algorithm is an experimental feature and might change at any time
```
Comment (RamIdeas) [inline packages/wrangler/e2e/__snapshots__/pages-dev.test.ts.snap:40]: The number in `(node:75279) ...` changes on every run which will cause the snapshots to fail

This line isn't in the other snapshot. Need to figure out where it's coming from

Comment (RamIdeas) [reply packages/wrangler/e2e/__snapshots__/pages-dev.test.ts.snap]: I just needed to add `--no-warnings` to the wrangler command used in my e2e command

File: packages/wrangler/src/dev/start-server.ts [hunk 5]
```diff
@@ -135,6 +135,9 @@ export async function startDevServer(
 			format: props.entry.format,
 			moduleRoot: props.entry.moduleRoot,
 		},
+		legacy: {
+			enablePagesAssetsServiceBinding: props.enablePagesAssetsServiceBinding,
+		},
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev/start-server.ts:140]: I think this is OK to put in legacy for now. But we might need to revisit this when we get Assets for Workers up and running.

File: packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts [hunk 6]
```diff
@@ -92,6 +92,23 @@ async function convertToConfigBundle(
 		event.bundle = { ...event.bundle, modules: [] };
 	}
 
+	if (event.config.legacy?.enablePagesAssetsServiceBinding !== undefined) {
```
Comment (petebacondarwin) [inline packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts:95]: should this be `==true`?

```suggestion
	if (event.config.legacy?.enablePagesAssetsServiceBinding === true) {
```

**Review by petebacondarwin — APPROVED** [review]
LGTM with a few nits.

File: packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts (hunk 6 above)
Comment (petebacondarwin) [reply packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts:95]: since setting that flag to false would pass this if test

Comment (RamIdeas) [reply packages/wrangler/src/api/startDevWorker/LocalRuntimeController.ts]: This wasn't a boolean flag. It was an object which if set opted into using the binding with the object as config.

I realised this binding, when resolved, ended up being just a fetcher which startWorker support as-is. So I've updated this PR to pass it in directly so startWorker has no knowledge of the PagesAssetsServiceBinding and no extra options to support

File: packages/wrangler/src/dev/start-server.ts (hunk 5 above)
Comment (RamIdeas) [reply packages/wrangler/src/dev/start-server.ts]: I've updated this PR to pass in a fetcher binding and therefore do not need to add this extra option to the API and support it forever

**Review by petebacondarwin — APPROVED** [review]
Still looks good. There are some flakes, perhaps?

**Pull request merged**